haystack/components/agents/agent.py

```diff
@@ -225,31 +226,34 @@ def run(
         **kwargs: Dict[str, Any],
     ) -> Dict[str, Any]:
         """
-        Process messages and execute tools until the exit condition is met.
+        Process messages and execute tools until an exit condition is met.
 
-        :param messages: List of chat messages to process
+        :param messages: List of Haystack ChatMessage objects to process.
+            If a list of dictionaries is provided, each dictionary will be converted to a ChatMessage object.
         :param streaming_callback: A callback that will be invoked when a response is streamed from the LLM.
             The same callback can be configured to emit tool results when a tool is called.
         :param kwargs: Additional data to pass to the State schema used by the Agent.
             The keys must match the schema defined in the Agent's `state_schema`.
-        :return: Dictionary containing messages and outputs matching the defined output types
+        :returns:
+            A dictionary with the following keys:
+            - "messages": List of all messages exchanged during the agent's run.
+            - "last_message": The last message exchanged during the agent's run.
```

Comment:
@bilgeyucel an important question, do you like `last_message` or would something like `final_message` be clearer?